pylearn2/datasets/hdf5.py

```diff
@@ -2,338 +2,472 @@
+                'and will become officially unsupported as of July 27, 2015. '
+                'The dataset should use the new interface that inherits from '
+                'the dataset class instead of the DenseDesignMatrix class. '
+                'Please refer to pylearn2.datasets.hdf5.py for more details '
+                'on arguments and details of the new '
+                'interface.', DeprecationWarning)
+            return HDF5DatasetDeprecated(filename, X, topo_view, y, load_all,
+                                         cache_size, **kwargs)
+        else:
+            return super(HDF5Dataset, cls).__new__(
+                cls, filename, sources, spaces, aliases, load_all, cache_size,
+                use_h5py, **kwargs)
 
-    def get_dataset(self, dataset, load_all=False):
+    def __init__(self, filename, sources, spaces, aliases=None, load_all=False,
+                 cache_size=None, use_h5py='auto', **kwargs):
         """
-        Get a handle for an HDF5 dataset, or load the entire dataset into
-        memory.
-
         Parameters
         ----------
-        dataset : str
-            Name or path of HDF5 dataset.
+        filename : str
+            HDF5 file name.
+        sources : list of str
+            A list of key(s) of dataset data in the HDF5 file.
+        spaces : list of Space objects
+            A list of spaces, one for each source in sources.
+        aliases : list of str, optional
+            A list of aliases, one for each source. They work as an alias for
+            the original name of the sources and once set can be used to access
+            the data. If you use this parameter, for each element in sources
+            you have to either specify an alias or None.
         load_all : bool, optional (default False)
-            If true, load dataset into memory.
+            If true, datasets are loaded into memory instead of being left
+            on disk.
+        cache_size: int, optional
+            This value is used when use_h5py is True.
+            The size in bytes for the chunk cache of the HDF5 library. Useful
+            when the HDF5 files has large chunks and when using a sequential
+            iterator. The chunk cache allows only access the disk for the
+            chunks and then copy the batches to GPU from memory, which can
+            result in a significant speed up. Sensible default values depend
+            on the size of your data and the batch size you wish to use. A
+            rule of thumb is to make a chunk contain 100 - 1000 batches and
+            make sure they encompass complete samples.
+        use_h5py: bool or 'auto', optional
+            Specifies if h5py or pytables should be used. If set to auto
+            pylearn2 will try to use pytables and will switch to h5py if
+            pytables cannot be loaded (e.g. is not installed).
+        kwargs : dict, optional
+            Keyword arguments passed to `DenseDesignMatrix`.
```

Comment:
That docstring should be directly at the class-level, not in the `__init__` method, otherwise it will not get properly displayed online.